src/rpdk/core/jsonutils/utils.py

```diff
@@ -111,6 +112,79 @@ def traverse(document, path_parts):
+
+    Differing from traverse, this returns a list of documents and a list of resolved paths.
+
+    :raises ValueError, LookupError: the reference is invalid for this document
+
+    >>> traverse_path_for_sequence_members({"foo": {"bar": [42, 43, 44]}}, tuple())
+    ([{'foo': {'bar': [42, 43, 44]}}], [()])
+    >>> traverse_path_for_sequence_members({"foo": {"bar": [42, 43, 44]}}, ["foo"])
+    ([{'bar': [42, 43, 44]}], [('foo',)])
+    >>> traverse_path_for_sequence_members({"foo": {"bar": [42, 43, 44]}}, ("foo", "bar"))
+    ([[42, 43, 44]], [('foo', 'bar')])
+    >>> traverse_path_for_sequence_members({"foo": {"bar": [42, 43, 44]}}, ("foo", "bar", "*"))
+    ([42, 43, 44], [('foo', 'bar', 0), ('foo', 'bar', 1), ('foo', 'bar', 2)])
+    >>> traverse_path_for_sequence_members({"foo": {"bar": [{"baz": 1, "bin": 1}, {"baz": 2, "bin": 2}]}}, ("foo", "bar", "*"))
+    ([{'baz': 1, 'bin': 1}, {'baz': 2, 'bin': 2}], [('foo', 'bar', 0), ('foo', 'bar', 1)])
+    >>> traverse_path_for_sequence_members({"foo": {"bar": [{"baz": 1, "bin": 1}, {"baz": 2, "bin": 2}]}}, ("foo", "bar", "*", "baz"))
+    ([1, 2], [('foo', 'bar', 0, 'baz'), ('foo', 'bar', 1, 'baz')])
+    >>> traverse_path_for_sequence_members({}, ["foo"])
+    Traceback (most recent call last):
+    ...
+    KeyError: 'foo'
+    >>> traverse_path_for_sequence_members([], ["foo"])
+    Traceback (most recent call last):
+    ...
+    ValueError: invalid literal for int() with base 10: 'foo'
+    >>> traverse_path_for_sequence_members([], [0])
+    Traceback (most recent call last):
+    ...
+    IndexError: list index out of range
+    """
+    if path is None:
+        path = []
+    if not path_parts:
+        return [document], [tuple(path)]
+    path_parts = list(path_parts)
+    if not isinstance(document, Sequence):
+        part = path_parts.pop(0)
+        document = document[part]
+        path.append(part)
+        return traverse_path_for_sequence_members(document, path_parts, path)
+    sequence_part = path_parts.pop(0)
+    if sequence_part == UNPACK_SEQUENCE_IDENTIFIER:
+        # if we should unpack the sequence, we need to grab multiple paths
+        documents = []
+        resolved_paths = []
+        new_documents = []
+        new_paths = []
+        for sequence_index in range(len(document)):
+            new_paths.append(path.copy() + [sequence_index])
+            new_document = traverse_path_for_sequence_members(
+                document, [sequence_index] + path_parts, path.copy()
+            )[0]
+            new_documents.extend(new_document)
+        for i in range(len(new_documents)):  # pylint: disable=consider-using-enumerate
+            new_document = new_documents[i]
+            newer_documents, newer_paths = traverse_path_for_sequence_members(
+                new_document, path_parts, new_paths[i]
+            )
+            documents.extend(newer_documents)
+            resolved_paths.extend(newer_paths)
+        return documents, resolved_paths
```

Comment:
Thoughts on this bit? I was looking into refactoring this bit, but I could not figure out a way to do so without re-writing `traverse_path_for_sequence_members` to support multiple documents as input arguments, and the added complexity was not worth it IMHO.

Reply:
I agree, supporting multiple documents as input arguments does not sounds right in this case